frontend/packages/topology/src/components/workload/pods-tab-section.tsx

```diff
@@ -24,9 +24,17 @@ const PodsTabSection: React.FC<{
     loaded: boolean;
   }>({ loaded: false });
 
-  const handleAdapterResolved = React.useCallback((data) => {
-    setPodData({ data, loaded: true });
-  }, []);
+  const handleAdapterResolved = React.useCallback(
+    (data) => {
+      if (podAdapter?.resource?.kind === 'CronJob') {
+        // Fixes the issue of Topology page crashing.
+        setTimeout(() => setPodData({ data, loaded: true }), 3000);
```

Comment:
why 3 seconds? would this timeout be enough if the user had slow/unreliable Internet speeds?

Reply:
This delay is created so that the client doesn't start watching the cluster immediately to get the pods for the job. Rather, this delay will give some time to the cluster to create the pods for the jobs. 

Normally it's seen that 3 secs is enough to create at least 15 pods in a cluster. Usually, Jobs can have around 5 pods, so we are good for most of the use cases.
Therefore, it is not directly involved with the user's internet connection, but rather the speed of the K8s cluster to create the pods.

But if we don't create this delay, then the client will be firing the useEffect hook every time a new pod is created finally leading to a page crash.

Reply:
what happens if k8s takes more than 3 seconds, e.g., when the server is under high load? would it try again after another 3 seconds?

Reply:
In that case, the page wouldn't crash at all. The page is crashing because the pods are created very fast causing rendering issue.

Reply:
I mean, I'm not involved in day to day issues anymore, but this looks really hacky to me.

I agree with @logonoff. We should try to fix so the root cause that raises a crash instead of solving the syntom. I expect that fast rerenderings could happen also in other scenarios then CronJobs.

It's good to have a a quick work around. I would recommend anyway to create a tech debt issue where someone goes deeper into that issue, solve the crash and then removes this 3s delay here as well.

@Lucifergene @vikram-raj